h5py/_hl/filters.py

```diff
@@ -259,7 +263,23 @@ def rq_tuple(tpl, name):
 
     if chunks is not None:
         plist.set_chunk(chunks)
-        plist.set_fill_time(h5d.FILL_TIME_ALLOC)  # prevent resize glitch
+
+    if fill_time is None:
+        if chunks is None:
+            fill_time = 'ifset'
+        else:
+            fill_time = 'alloc'  # prevent resize glitch
+    else:
+        invalid_fill_time = (isinstance(fill_time, str) and
+                             fill_time.lower() not in
+                             ('alloc', 'never', 'ifset'))
+        fill_time_not_str = not isinstance(fill_time, str)
```

Comment:
If we're going to the trouble of checking the type separately, we could raise a TypeError. Or we could keep the code simple and just check `fill_time in _FILL_TIME_ENUM`. :slightly_smiling_face:

Reply:
Yes I agree with the simple check `fill_time in _FILL_TIME_ENUM`. I think I was trying to prevent users passing the enum value directly and somehow guided by some type check...